lib/storage/data/file/DataFileStore.js

```diff
@@ -178,6 +184,18 @@ class DataFileStore {
                     return cbOnce(null, key);
                 }
                 if (this.noSync) {
+                    /*
+                     * It's is not guaranteed that the Kernel will release page
+                     * caches when this.noSync is true. If you want to ensure
+                     * this behavior, set this.noSync to false.
+                     */
+                    if (this.noCache) {
```

Comment:
Now that the code executed post-fsync becomes non-trivial I would recommend creating a helper function to avoid code duplication.